lib/collection/src/wal.rs

```diff
@@ -110,8 +110,16 @@ impl<'s, R: DeserializeOwned + Serialize + Debug> SerdeWal<R> {
             .map_err(|err| WalError::WriteWalError(format!("{err:?}")))
     }
 
-    pub fn read_all(&'s self) -> impl Iterator<Item = (u64, R)> + 's {
-        self.read(self.first_index())
+    pub fn read(&self, start_index: u64) -> impl DoubleEndedIterator<Item = (u64, R)> + '_ {
```

Comment:
I like that we use `DoubleEndedIterator` here now so we have a single function. But I must admit, this way I find `start_index` very confusing.

I'd rather have a `read_all(with_acknowledged: bool)` which is also doubly ended to prevent any confusion here.

I'd strongly vote for the latter because it prevents exposing unnecessary implementation details - such as record indices - to the public.

Reply:
I think I have to agree now. I'll return the `read_all` method back. I also don't like these indices, cause they are confusing as hell.